### PR TITLE
macropad rev B definion and fix for SH110x addressing 

### PR DESCRIPTION
**Pull request by ladyada**
the SH1107 chipset is actually column not row mode (like the SSD1306!) BUT the SH1107 *module* we use on the featherwing is vertical orientation (confusing!). We should stick to the column mode addressing, but 'rotate' the SH1107 module. 

we will have to fix the SH1107 driver to match

this fix required for SH1106 (which uses the same page mode commands but ISNT rotated). Tested on Macropad hardware

### CODE REVIEW

**Review by tannewt — COMMENTED** [review]
Have you tried rotation with this? Seems like it'd be broken before and after this change.

Could you link me to the display datasheets?

**Comment by ladyada** [comment]
[SH1106.pdf](https://github.com/adafruit/circuitpython/files/6612400/SH1106.pdf)

**Comment by ladyada** [comment]
[SH1107.pdf](https://github.com/adafruit/circuitpython/files/6612401/SH1107.pdf)

**Comment by ladyada** [comment]
i will try rotation now (i also have to try the col offset thing)

File: shared-module/displayio/Display.c [hunk 1]
```diff
@@ -249,8 +249,8 @@ STATIC bool _refresh_area(displayio_display_obj_t *self, const displayio_area_t
     // for SH1107 and other boundary constrained controllers
     //      write one single row at a time
     if (self->SH1107_addressing) {
-        subrectangles = rows_per_buffer;  // vertical (column mode) write each separately (height times)
-        rows_per_buffer = 1;
+        subrectangles = rows_per_buffer / 8;  // vertical (column mode) write each separately (height times)
```
Comment (tannewt) [inline shared-module/displayio/Display.c:252]:
```suggestion
        subrectangles = rows_per_buffer / 8;  // page addressing mode writes 8 rows at a time
```

**Review by tannewt — CHANGES_REQUESTED** [review]
One minor thing plus the colstart and rowstart fix.

**Review by tannewt — APPROVED** [review]
Looks good to me! Go ahead and merge once CI gives the OK.

**Pull request merged**